lib/lotus/model/adapters/sql/command.rb

```diff
@@ -55,10 +59,41 @@ def update(entity)
           # @api private
           # @since 0.1.0
           def delete
-            @collection.delete
+            _rescue_database_error do
+              @collection.delete
+            end
           end
 
           alias_method :clear, :delete
+
+          private
+
+          # Rescues Sequel::DatabaseError in yielded block
+          #
+          # @api private
+          def _rescue_database_error
+            yield
+          rescue Sequel::DatabaseError => e
+            raise _transform_database_error(e)
```

Comment:
@Moratorius I'd go for:

```ruby
raise Lotus::Model::Error.new(e.message)
```

That's it.

Starting from Ruby 2.1, exceptions respond to `#cause`, which returns the original raised exception. For our case, our `Lotus::Model::Error#cause` will return one of the `Sequel` raised exceptions. Read more at: http://devblog.avdi.org/2013/12/25/exception-causes-in-ruby-2-1/